examples/fabric-admin/device_manager/DeviceManager.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DeviceManager.h"
+
+#include <commands/interactive/InteractiveCommands.h>
+
+#include <cstdio>
+#include <string>
+
+using namespace chip;
+using namespace chip::app::Clusters;
+
+namespace {
+
+// Constants
+constexpr uint32_t kSyncPrepareTimeMs = 1000;
+
+} // namespace
+
+// Define the static member
+DeviceManager DeviceManager::sInstance;
+
+void DeviceManager::Init()
```

Comment:
nit, should we add a mechanism to make sure `Init` is called before any other method in `DeviceManager`? If that is overkill maybe a comment as to why we didn't do that?

Reply:
acked